[packaging] simpler dependency plotting

zdevito opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45686 [packaging] simpler dependency plotting**
* #45471 [fx] import used modules for code gen

This uses an online graphviz viewer. The code is simpler, and
since it embeds all the data in the url you can just click the url
from your terminal.

Differential Revision: [D24059157](https://our.internmc.facebook.com/intern/diff/D24059157)

zdevito committed:
[packaging] simpler dependency plotting
This uses an online graphviz viewer. The code is simpler, and
since it embeds all the data in the url you can just click the url
from your terminal.

[ghstack-poisoned]

suo commented on test/test_package.py:
```diff
@@ -186,8 +186,9 @@ def test_resnet(self):
-            e._write_dep_graph(failing_module='torch', output_file=buf)
-            self.assertIn('torchvision.models.resnet', buf.getvalue())
+            debug_graph = e._write_dep_graph(failing_module='torch')
+            print(debug_graph)
```
stray print

suo reviewed (approved):
nice! Could be nice as a followup to directly return the graphviz string, since there are CLI tools and such that can visualize.

zdevito committed:
Update on "[packaging] simpler dependency plotting"


This uses an online graphviz viewer. The code is simpler, and
since it embeds all the data in the url you can just click the url
from your terminal.

Differential Revision: [D24059157](https://our.internmc.facebook.com/intern/diff/D24059157)

[ghstack-poisoned]

zdevito committed:
Update on "[packaging] simpler dependency plotting"


This uses an online graphviz viewer. The code is simpler, and
since it embeds all the data in the url you can just click the url
from your terminal.

Differential Revision: [D24059157](https://our.internmc.facebook.com/intern/diff/D24059157)

[ghstack-poisoned]

zdevito committed:
Update on "[packaging] simpler dependency plotting"


This uses an online graphviz viewer. The code is simpler, and
since it embeds all the data in the url you can just click the url
from your terminal.

Differential Revision: [D24059157](https://our.internmc.facebook.com/intern/diff/D24059157)

[ghstack-poisoned]
